tests/system_tests_http1_observer.py

```diff
@@ -0,0 +1,240 @@
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+#
+# Test the HTTP/1.x Protocol observer
+#
+
+import os
+
+from system_test import TestCase, unittest, main_module, Qdrouterd
+from system_test import curl_available, run_curl
+from system_test import nginx_available, NginxServer
+from system_test import Http1Server, retry, TIMEOUT
+from vanflow_snooper import VFlowSnooperThread
+
+
+def spawn_nginx(port, tester):
+    """
+    Spawn the Nginx server listening on port
+    """
+    env = dict()
+    env['nginx-base-folder'] = NginxServer.BASE_FOLDER
+    env['setupclass-folder'] = tester.directory
+    env['nginx-configs-folder'] = NginxServer.CONFIGS_FOLDER
+    env['listening-port'] = str(port)
+    env['http2'] = ''  # disable HTTP/2
+    # disable ssl/tls for now
+    env['ssl'] = ''
+    env['tls-enabled'] = '#'  # Will comment out TLS configuration lines
+
+    # TBD: TLS stuff
+    # env['ssl'] = 'ssl'
+    # env['tls-enabled'] = ''  # Will enable TLS lines
+    # env['chained-pem'] = CHAINED_CERT
+    # env['server-private-key-no-pass-pem'] = SERVER_PRIVATE_KEY_NO_PASS
+    # env['ssl-verify-client'] = 'on'
+    # env['ca-certificate'] = CA_CERT
+
+    return tester.nginxserver(config_path=NginxServer.CONFIG_FILE, env=env)
+
+
+@unittest.skipUnless(nginx_available() and curl_available(),
+                     "Requires both nginx and curl tools")
+class Http1ObserverTest(TestCase):
+    """
+    Verify the HTTP/1.x observer produces the expected VanFlow records
+    """
+    @classmethod
+    def router(cls, name, listener_port, server_port, extra_config=None):
+        """
+        Create a router with a tcpConnector and a tcpListener. Set the logging
+        config to only enable Van Flow record logs.
+        """
+        config = [
+            ('router', {'mode': 'interior',
+                        'id': name}),
+            ('listener', {'role': 'normal',
+                          'port': cls.tester.get_port()}),
+
+            ('tcpListener', {'host': "0.0.0.0",
+                             'port': listener_port,
+                             'address': 'Http1ObserverTest'}),
+            ('tcpConnector', {'host': "localhost",
+                              'port': server_port,
+                              'address': 'Http1ObserverTest'}),
+
+            ('log', {'module': 'DEFAULT',
```

Comment:
`Qdrouterd` class in system_test.py already enabled `debug+` for all log modules. Is this necessary ?

Reply:
Ah - no it's not.  Originally I had the test parse the log output looking for van flow log events to validate the observer. I changed that to use the external vanflow_snooper instead. I'll remove that log config since it's not necessary.

Reply:
Done